datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -1349,10 +1349,18 @@ private void handleLeaderDoAssignment(boolean cleanUpOrphanNodes) {
         failedStreams = unclaimedTokens.keySet();
```

Comment:
Do you think we should be handling this better? When a stop request fails, the datastream will be stuck in the STOPPING state. Since the leader will update the ZK with the newer assignment (removing the stopping tasks) and would also clean up the _liveTaskMap of the zkAdapter, a subsequent stopping request would not be handled properly since the leader would not have the host to task assignment for that stuck STOPPING state datastream to issue tokens for the hosts hosting tasks for stopping datastream. Thus the stream would forever be stuck in the STOPPING state.

I am trying to think if we can somehow recover from this in code, like should we at least somehow maintain a state where we can retrigger the stop request so that the leader could reattempt to stop this datastream.

Reply:
Added a section regarding this in the supporting document. Setting up a meeting to discuss this.

Reply:
Force stop path was added as a recovery for stream stuck in STOPPING state. Also a configuration is added that stops the streams even if stop failed/timed out on some nodes. Both of these mechanisms are interim solutions to avoid any regressions. A more comprehensive solution to be implemented as part of Phase 2 (a restart endpoint).